src/market/directory.rs

```diff
@@ -123,12 +125,18 @@ impl DirectoryServer {
         let mut flag = self
             .shutdown
             .write()
-            .map_err(|_| DirectoryServerError::Other("Rwlock write error!"))?;
+            .map_err(|_| DirectoryServerError::PoissonError("Rwlock write error!"))?;
```

Comment:
Won't need the map_err if use the suggested pattern.

Reply:
Agreed.